californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -546,6 +546,18 @@ public void receiveResponse(final Exchange exchange, final Response response) {
 					upper().receiveResponse(exchange, resp);
 				}
 			} else if (!response.hasBlockOption()) {
+				
+				// most recent none blockwise notification should stop current blockwise transfer
+				KeyUri key = getKey(exchange, response);
+				Block2BlockwiseStatus status = getBlock2Status(key);
+				if (response.getOptions().hasObserve() && status != null && status.isNotification()
+						&& status.getObserve() < response.getOptions().getObserve()) {
+					// complete current exchange
+					exchange.setComplete();
+					exchange.setCurrentRequest(exchange.getRequest());
+					// clear blockwise status
+					clearBlock2Status(key);
```

Comment:
Which part give you some concerns  ?
ClearBlock ?
ClearExchange ?
or both? 

Personally, I think the clearblock2 make senses and is useful.
The exchange one is not so clear mainly because of the design and not so useful as I know one of your previous fix resolve the cleaning exchange indirectly, this is just the "explicit" way.
I think we need a clearer way to clean exchange when we want to interrupt a blockwise transfer.